Automattic/wp-calypso
# Add a unified Plans step that is completely shared between Stepper and Start

Context: pdDR7T-1Ed-p2#comment-2179

Depends on https://github.com/Automattic/wp-calypso/pull/93035

## Proposed Changes

This aims to create the thinnest wrapper possible to wrap the Plans step in /start and have it in Stepper as well, to achieve parity without repeating any work. 

## Why are these changes being made?
To be able to delete `client/landing/stepper/declarative-flow/internals/steps-repository/plans` and have a single Plans step across all frameworks and places. And to have all the improvements that go into the plans step shared between Stepper and Start.

## Testing Instructions

1. Go to `/setup/onboarding`.
2. Play with the plans step.
3. It should be identical to the one at `/start/plans`.
4. If you select a free domain, it should show the free plan is a column. **And should not have a subheader.**
5. If you pick a paid domain, it should show the free plan de-emphasized **in the subheader**.
6. Clicking the subheader should show a dialog. 
7. Picking a plan from the dialog should land you checkout with the correct plan.
8. Pick "Continue with free domain" should remove the paid domain and land you in site-setup flow.

In general, it should be an exact UX replica of /start/plans.
